command/agent/cache/lease_cache.go

```diff
@@ -994,6 +1020,25 @@ func (c *LeaseCache) restoreLeases(leases [][]byte) error {
 		if err != nil {
 			return err
 		}
+
+		// Check if this lease has already expired
+		reader := bufio.NewReader(bytes.NewReader(newIndex.Response))
+		resp, err := http.ReadResponse(reader, nil)
+		if err != nil {
+			c.logger.Error("failed to deserialize response", "error", err)
+			return err
+		}
+		secret, err := api.ParseSecret(resp.Body)
+		if err != nil {
+			c.logger.Error("failed to parse response as secret", "error", err)
+			return err
+		}
+		elapsed := time.Now().Sub(newIndex.LastRenewed)
+		if int(elapsed.Seconds()) > secret.LeaseDuration {
```

Comment:
For auth leases you might have to check `secret.Auth.LeaseDuration` instead.

Reply:
Added in b3c32c1ec918d083a795b882c656f773b84604ba